src/main/java/edu/harvard/iq/dataverse/AuxiliaryFileServiceBean.java

```diff
@@ -0,0 +1,114 @@
+
+package edu.harvard.iq.dataverse;
+
+import edu.harvard.iq.dataverse.dataaccess.StorageIO;
+import edu.harvard.iq.dataverse.util.FileUtil;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Logger;
+import javax.ejb.EJB;
+import javax.ejb.Stateless;
+import javax.inject.Named;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import org.apache.tika.Tika;
+
+/**
+ *
+ * @author ekraffmiller
+ *  Methods related to the AuxiliaryFile Entity.
+ */
+@Stateless
+@Named
+public class AuxiliaryFileServiceBean implements java.io.Serializable {
+   private static final Logger logger = Logger.getLogger(AuxiliaryFileServiceBean.class.getCanonicalName());
+
+    @PersistenceContext(unitName = "VDCNet-ejbPU")
+    private EntityManager em;
+    
+    @EJB
+    private SystemConfig systemConfig;
+    
+
+    public AuxiliaryFile find(Object pk) {
+        return em.find(AuxiliaryFile.class, pk);
+    }
+
+    public AuxiliaryFile save(AuxiliaryFile auxiliaryFile) {
+        AuxiliaryFile savedFile = em.merge(auxiliaryFile);
+        return savedFile;
+
+    }
+    
+    /**
+     * Save the physical file to storageIO, and save the AuxiliaryFile entity
+     * to the database.  This should be an all or nothing transaction - if either
+     * process fails, than nothing will be saved
+     * @param fileInputStream - auxiliary file data to be saved
+     * @param dataFile  - the dataFile entity this will be added to
+     * @param formatTag - type of file being saved
+     * @param formatVersion - to distinguish between multiple versions of a file
+     * @param origin - name of the tool/system that created the file
+     * @param isPublic boolean - is this file available to any user?
+     * @return success boolean - returns whether the save was successful
+     */
+    public boolean processAuxiliaryFile(InputStream fileInputStream, DataFile dataFile, String formatTag, String formatVersion, String origin, boolean isPublic) {
+    
+        StorageIO<DataFile> storageIO =null;
+      
+        String auxExtension = formatTag + "_" + formatVersion;
+        try {
+            // Save to storage first.
+            // If that is successful (does not throw exception),
+            // then save to db.
+            // If the db fails for any reason, then rollback
+            // by removing the auxfile from storage.
+            storageIO = dataFile.getStorageIO();
+            AuxiliaryFile auxFile = new AuxiliaryFile();
+            storageIO.saveInputStreamAsAux(fileInputStream, auxExtension);
+            auxFile.setChecksum(FileUtil.calculateChecksum(storageIO.getAuxFileAsInputStream(auxExtension), systemConfig.getFileFixityChecksumAlgorithm()));
+
+            Tika tika = new Tika();
```

Comment:
So far Tika is only used for full-text indexing and content-type detection is done differently for Datafiles. Could/should these both be done the same way? (not necessarily in this PR)

Reply:
for Datafiles, JHove is used, which works on a file as opposed to an inputStream.  So to use JHove, I think I would have to save the aux file to a temp location, unless I'm missing something. Is this something we want to do?

Reply:
FWIW: Requiring a temp file for mime detection is a problem for Datafiles too, so it may make sense to use Tika instead of JHOVE there as well. (#6937 would make it possible to get ranges of bytes from S3 to help avoid having to get a whole file).

Reply:
ok, should we create a separate ticket for changing mime detection to Tika for Datafiles? (Is this something that is usually talked about at dv_tech hour?)

Reply:
Discussed this during tech hours, we agreed that it would be good to eventually switch from JHove to Tika in other parts of the code for consistency

Reply:
I'm about to approve the PR. There's just one thing I regret not having thought about earlier, when we were having this discussion, above: WHY are we trying to detect mime types at all? As opposed to just making the uploading client supply it, as another parameter? 
We try to detect mime types on "normal" uploaded files - but that's because users upload  arbitrary files... This API is for something that's structured; at least in the immediate use case scenario. I.e. that preprocessed summary stats fragment will always be in JSON, the diff. private DDI will always be XML etc. etc. We don't expect any variety there... 
I don't see this as  a problem, having this detection code in place. It may come in handy for other cases. And, even if we decide that these aux uploads should, or can supply the mime type as a parameter - we could use this detection as an extra validation step. 
Still, I can't help but feel I should've said "let's not even worry about it, let's just add a parameter instead", early on.

Reply:
I'm about to approve the PR. There's just one thing I regret not having thought about earlier, when we were having this discussion, above: WHY are we trying to detect mime types at all? As opposed to just making the uploading client supply it, as another parameter?
We try to detect mime types on "normal" uploaded files - but that's because users upload arbitrary files... This API is for something that's structured; at least in the immediate use case scenario. I.e. that preprocessed summary stats fragment will always be in JSON, the diff. private DDI will always be XML etc. etc. We don't expect any variety there...
I don't see this as a problem, having this detection code in place. It may come in handy for other cases. And, even if we decide that these aux uploads should, or can supply the mime type as a parameter - we could use this detection as an extra validation step.
Still, I can't help but feel I should've said "let's not even worry about it, let's just add a parameter instead", early on.